wp-content/plugins/wdgrestapi/entities/sendinblue-template.php

```diff
@@ -0,0 +1,147 @@
+	public static $entity_type = 'sendinblue_template';
+
+	public function __construct( $slug = FALSE ) {
+		parent::__construct( FALSE, self::$entity_type, self::$db_properties );
+
+		if ( !empty( $slug ) ) {
+			global $wpdb;
+			$table_name = WDGRESTAPI_Entity::get_table_name( self::$entity_type );
+			$query = "SELECT * FROM " .$table_name. " WHERE slug = '" .$slug. "'";
+			$row_data = $wpdb->get_row( $query );
+			if ( !empty( $row_data ) ) {
+				$this->loaded_data = $row_data;
+			}
+		}
+	}
+
+	public static function get_by_fr_id( $fr_id ) {
+		global $wpdb;
+		if ( empty( $wpdb ) ) {
+			return FALSE;
+		}
+		$table_name = WDGRESTAPI_Entity::get_table_name( self::$entity_type );
+		$query = 'SELECT * FROM ' .$table_name. ' WHERE id_sib_fr=' .$fr_id;
+		$result = $wpdb->get_row( $query );
+		if ( empty( $result ) || empty( $result->id ) ) {
+			return FALSE;
+		}
+		$user = new WDGRESTAPI_Entity_SendinblueTemplate( $result->slug );
+		return $user;
+	}
+
+	/**
+	 * Met à jour les textes à la fois sur SendInBlue et sur Transifex, selon les avancées des uns et des autres
+	 */
+	public function update_texts() {
+		// On commence par récupérer les chaines traduites (car si on met à jour Transifex d'abord, ça les écrasera)
+		$this->update_language_content_from_transifex( 'en' );
+		// Puis on renvoie les données du template français vers Transifex
+		$this->send_french_content_to_transifex();
+	}
+
+	/**
+	 * Retourne l'objet SIB
+	 */
+	private function get_sendinblue() {
+		include_once( plugin_dir_path( __FILE__ ) . '../libs/sendinblue/mailin.php');
+		$mailin = new Mailin( 'https://api.sendinblue.com/v2.0', WDG_SENDINBLUE_API_KEY, 8000 );
+		return $mailin;
+	}
+
+	/**
+	 * Va chercher le contenu du template sur SiB
+	 */
+	public function get_french_content() {
+		$mailin = $this->get_sendinblue();
+		$data = array( 'id' => $this->loaded_data->id_sib_fr );
+		$sendinblue_result = $mailin->get_campaign_v2( $data );
+		return $sendinblue_result[ 'data' ][ 0 ][ 'html_content' ];
+	}
+
+	/**
+	 * Envoie le contenu français (d'origine) vers Transifex
+	 */
+	public function send_french_content_to_transifex() {
+		$french_content = $this->get_french_content();
+		WDGRESTAPI_Lib_Transifex::create_or_update_resource( $this->loaded_data->slug, $french_content, 'Template ' .$this->loaded_data->slug );
+	}
+
+	/**
+	 * Envoie le contenu traduit sur Transifex vers SendInBlue
+	 */
+	public function update_language_content_from_transifex( $language_code ) {
+		// Récupération du contenu traduit sur Transifex
+		$translation_content = WDGRESTAPI_Lib_Transifex::get_resource_translation( $this->loaded_data->slug, $language_code );
+		//echo '<br><br>translation_content :<br>';
+		//print_r( $translation_content );
+		if ( empty( $translation_content ) ) {
+			return FALSE;
+		}
+
+		$mailin = $this->get_sendinblue();
+
+		// Si il n'y a pas encore de template traduit, on le crée
+		if ( empty( $this->loaded_data->id_sib_en ) ) {
```

Comment:
@emilschn : empty( $this->loaded_data->id_sib_en ) --> empty( $this->loaded_data['id_sib_"+$language_code] )